src/intrinsics/ecma262/Object.js

```diff
@@ -101,36 +136,69 @@ export default function(realm: Realm): NativeFunctionValue {
         }
```

Comment:
When we call `frm.makeNotPartial()` we assume that we will call `frm.makePartial()` after that but if we throw and then recover later, we assume a lot about that the reversal process keeps working like it does today. I don't see an immediate bug right now but I see a ton of subtle changes we could do that would cause this to be a bug. I wonder if there's a way we can make the guarantees of resetting this to partial be stronger. E.g. a try/finally around the place we need it `frm.$OwnPropertyKeys()`.

Reply:
Done